From b0be318edc9c9de1d504555c4338364dacd6896e Mon Sep 17 00:00:00 2001
From: Florian Dollinger <atar-axis@users.noreply.github.com>
Date: Mon, 15 Apr 2019 08:59:36 +0200
Subject: [PATCH 05/29] Check for final assignments in RUN keys (#12309)

As described in #12291

(cherry picked from commit 5a1ee0730206877a34a80a030db27dac2de38974)
---
 src/udev/udev-rules.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/udev/udev-rules.c b/src/udev/udev-rules.c
index ee87d7c65c..63a967633b 100644
--- a/src/udev/udev-rules.c
+++ b/src/udev/udev-rules.c
@@ -2442,6 +2442,11 @@ int udev_rules_apply_to_event(
                 case TK_A_RUN_BUILTIN:
                 case TK_A_RUN_PROGRAM: {
                         _cleanup_free_ char *cmd = NULL;
+                        
+                        if (event->run_final)
+                                break;
+                        if (cur->key.op == OP_ASSIGN_FINAL)
+                                event->run_final = true;
 
                         if (IN_SET(cur->key.op, OP_ASSIGN, OP_ASSIGN_FINAL))
                                 ordered_hashmap_clear_free_key(event->run_list);
-- 
2.21.0

